general/pull-request.md

```diff
@@ -0,0 +1,250 @@
+- Deploy `develop` to [testing environment]
+- Endorse to QA with 7 characters commit hash
+- Wait for QA results
+
+#### Bug fixing in Testing Phase
+
+- Estimate time needed to address bugs
+- Report estimated time to QA
+- Go back to Development Phase step
+
+### Deployment Phase
+- Adjust version files in repository
+
+Node: in `package.json` - `version` field
+
+Go: in `template.yml` - `Metadata.SemanticVersion` field
+
+Rails: `.version` file
+
+- Commit and push changes
+- Create PRs for version change to both `main` and `develop` branches
+- Wait for `main` PR approval
+- create a release (in github)
+- create tag (Note: don't put `v` prefix)
+- Deploy tag created to production
+
+### Post-deployment phase
+
+## PR Time Period
+
+- PRs should be resolved before the estimated Time commitment in ticket (ideally: the day before QA testing phase)
+
+## Code Reviewers
+
+Note: All are required unless stated
+
+### PR to `develop`
+
+- All *present* members of vertical team
+- 1 Horizontal lead - If template.yml/database structure changed (BE)
+- 1 Horizontal lead - Site/Screen structure (FE)
+- 1 representative from other verticals (ideally junior)
+
+Note:
+Liability of Vertical Lead + Sub lead/s if branch merged to `develop` caused errors
+
+### PR to `main`
+
+- Same as above
+- CTO/Engineering Manager
+- 2 Horizontal Leads - If CTO/Engineering Manager is not available
+- Other Technical leads if No Horizontal leads available
+
+Note:
+Liability of Vertical Lead + Sub lead/s + Approvers if `main` branch is broken
+
+
+
+## Branch Naming Guide
+
+- name pattern `prefix/context`
+- All characters should be lowercase
+- use `kebab-case` naming
+- descriptive name but not too long
+- **Max characters:** `15` (not counting prefix)
+
+
+
+Use the following prefixes for branch names:
+
+### `feat/`
+
+for code changes that provide new features for the application
+
+PR this branch to `develop` branch
+
+examples:  
+- New endpoints
+- New web pages/app screens
+- New sections in the app frontend
+
+
+### `fix/`
+
+for code changes that provides fixes for bugs found in the application
+
+PR this branch to `develop` branch
+
+
+### `chore/`
+
+for maintenance code in app
+
+PR this branch to `develop` branch
+
+
+examples:
+- updated dependencies
+- copy changes
+- asset changes
+- code refactors
+- deleting unused code/files
+
+### `release/`
+
+Merge `develop` branch changes to production
+
+branch name will be the version of the release ex: `release/1.2.3`
+
+PR this branch to `main` and `develop` branches
+
+
+### `hotfix/`
+
+- This must be checked out from `main` branch
+
+- branch name will be the version of the release ex: `hotfix/1.2.3`
+
+PR this branch to `main` and `develop` branches
+
+### `test/`
+
+For code changes that you want to test but not necessarily want to merge it (maybe for poc)
+
+Rename to proper branch if you want to merge
+
+
+
+
+## PR Description Outline
+
+- change description

```

Comment:
we can also encourage to specify here the business-related context/changes, apart from the technical/implementation-related changes